protoc-gen-swagger/genswagger/template.go

```diff
@@ -1448,6 +1459,88 @@ func processExtensions(inputExts map[string]*structpb.Value) ([]extension, error
 	return exts, nil
 }
 
+func validateHeaderType(headerType string) error {
+	switch headerType {
+	case
+		"string",
+		"number",
+		"integer",
+		"boolean":
+		return nil
+	}
+	return fmt.Errorf("the provided header type %q is not supported", headerType)
+}
+
+func validateDefaultValueType(headerType string, defaultValue string) error {
+	switch headerType {
+	case
+		"string":
+		if !(isString(defaultValue)) {
+			return fmt.Errorf("the provided default value %q does not match provider type %q, or is not properly quoted with escaped quotations", defaultValue, headerType)
+		}
+	case
+		"number":
+		_, err := strconv.ParseFloat(defaultValue, 64)
+		if err != nil {
+			return fmt.Errorf("the provided default value %q does not match provider type %q", defaultValue, headerType)
+		}
+	case
+		"integer":
+		_, err := strconv.ParseInt(defaultValue, 0, 64)
+		if err != nil {
+			return fmt.Errorf("the provided default value %q does not match provider type %q", defaultValue, headerType)
+		}
+	case
+		"boolean":
+		if !isBool(defaultValue) {
+			return fmt.Errorf("the provided default value %q does not match provider type %q", defaultValue, headerType)
+		}
+	}
+	return nil
+}
+
+func isString(s string) bool {
+	var digitCheck = regexp.MustCompile(regString)
+	return digitCheck.MatchString(s)
+}
+
+func isBool(s string) bool {
+	//Unable to use strconv.ParseBool because it returns truthy values https://golang.org/pkg/strconv/#example_ParseBool
+	// per https://swagger.io/docs/specification/data-models/data-types/#boolean
+	// type: boolean represents two values: true and false. Note that truthy and falsy values such as "true", "", 0 or null are not considered boolean values.
+	var digitCheck = regexp.MustCompile(regBool)
+	return digitCheck.MatchString(s)
+}
+
+func processHeaders(inputHdrs map[string]*swagger_options.Header) (swaggerHeadersObject, error) {
+	hdrs := map[string]swaggerHeaderObject{}
+	for k, v := range inputHdrs {
+		ret := swaggerHeaderObject{
+			Description: v.Description,
+			Format:      v.Format,
+			Pattern:     v.Pattern,
+		}
+		err := validateHeaderType(v.Type)
```

Comment:
Do we want to validate that the Format matches the Type as well? For instance integer and floating point sizes are specified in the format.

Reply:
The spec indicates: 
```
The extending format for the previously mentioned type. See Data Type Formats for further details.
```
https://github.com/OAI/OpenAPI-Specification/blob/3.0.0/versions/2.0.md#dataTypeFormat

Do we just want to validate the types as defined by the swagger spec?

![image](https://user-images.githubusercontent.com/19742213/96211105-ac63ca80-0f30-11eb-9f1e-7ee96dfbfa85.png)

Reply:
I think for each type we can check if the format is one of the specified and perform that validation, otherwise fallback to some permissive parsing (int64 for integer, float64 for number). I don't think we need to worry about the types we don't support (e.g. `string`, `password`). Lets just do `integer` (32 and 64), `number` (32 and 64), `string` and `boolean`.

Reply:
lol I guess that would be easier path than validating everything 🤣 

```
	case "number":
		if format != "" {
			switch format {
			case
				"uint",
				"uint8",
				"uint16",
				"uint32",
				"uint64",
				"int",
				"int8",
				"int16",
				"int32",
				"int64",
				"float",
				"float32",
				"float64",
				"complex64",
				"complex128",
				"double",
				"byte",
				"rune",
				"uintptr":
				return nil
			}
			return fmt.Errorf("the provided format %q is not a valid extension of the type %q", format, headerType)
		}
		return nil
```